### ISSUE TITLE
Standard scrollTo usage often results in errors

### ISSUE THREAD

**Issue by ctrlplusb** [cypress-io/cypress#751]
### Current behavior:

Running the example code, which performs a simple `scrollTo` often results in an error stating that multiple elements cannot be scrolled.

### Desired behavior:

Don't error, and scroll the page. 🤡

### How to reproduce:

Run the example code.

#### Test code:

```js
  context('Page Scroll', () => {
    it('should work', () => {
      cy.visit('https://www.theguardian.com/uk')
      cy.scrollTo('0%', '35%')
    })
  })
```

### Additional Info (images, stack traces, etc)

<img width="475" alt="screenshot 2017-10-17 16 03 22" src="https://user-images.githubusercontent.com/12164768/31672571-db28a682-b354-11e7-8e2e-d0e1c61ab7ac.png">

- Operating System: Mac OS 10.12.6
- Cypress Version: 1.0.2
- Browser Version: Chrome 61

**Comment by jennifer-shehane** [comment]
Code for this command can be found here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/actions/scroll.coffee

My first impression is that we are either:
1. Not testing this command on a website with multiple windows (iframes)
2. Not passing the window subject incorrectly to the scrollTo command.

**Comment by binaryst4r** [comment]
Any updates to this?

**Comment by omaotzu** [comment]
Also window.scrollTo isnt cross browser supported

https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo

**Comment by jennifer-shehane** [comment]
The implementation actually uses jQuery `scrollTo`, which I'd have to look back into to remember their implementation of.

**Comment by samuelmartineau** [comment]
@ctrlplusb  You could temporarily use vanilla js before having a real fix on `scrollTo`

```js
cy.window().then(win => {
  win.scrollTo('0%', '35%');
})
```

**Comment by jennifer-shehane** [comment]
I'm no longer able to get this code example working. Can anyone provide a basic example of this scroll bug?

**Comment by ca-pli** [comment]
This is because window.length is equal to the number of iframes on the page, causing above assertion error.

**Comment by Kjir** [comment]
I reproduced it easily in my project by just having a google map on the page: it will create an `iframe` and the previously existing `cy.scrollTo(0, 1);` suddenly breaks.
We worked around it by not displaying the map on the e2e tests, not ideal, but does the job...

**Comment by Kjir** [comment]
Here is a simple reproduction script:

```javascript
describe('ScrollTo with iframe', () => {
  it('does not scroll', () => {
    cy.visit('https://www.nytimes.com');
    cy.scrollTo('bottom');
  });
});
```

**Comment by jennifer-shehane** [comment]
Thanks @Kjir, I can recreate this again. 

<img width="497" alt="Screen Shot 2019-03-14 at 4 12 54 PM" src="https://user-images.githubusercontent.com/1271364/54346792-1856c080-4674-11e9-8517-81e0d59f0b95.png">
It may be trying to scroll all windows within the page, so errors when iframes. This is a complete guess though.

**Comment by jayeshhpl** [comment]
please fix this issue, spent hours finding for other solution

**Comment by jennifer-shehane** [comment]
The key to this issue was here: https://github.com/cypress-io/cypress/issues/751#issuecomment-465876618

>This is because window.length is equal to the number of iframes on the page, causing above assertion error if the page has iframe.

This was fixed in https://github.com/cypress-io/cypress/pull/4397 and released in [3.3.2](https://on.cypress.io/changelog#3-3-2)
